jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -226,44 +210,50 @@ public static boolean isValidJavaClass(String className) {
    * @throws NumberFormatException if any of the ports are in an invalid format or out of range
    */
   @VisibleForTesting
-  static ImmutableSortedMap<String, EmptyStruct> portListToPortMap(List<String> ports)
-      throws NumberFormatException {
-    SortedMap<String, EmptyStruct> result = new TreeMap<>();
+  static ImmutableList<String> expandPortRanges(List<String> ports, BuildLogger logger) {
+    List<String> result = new ArrayList<>();
 
     for (String port : ports) {
       // Remove all spaces
-      port = port.trim().replace(" ", "");
+      port = port.replace(" ", "");
 
       if (port.matches("\\d+")) {
         // Port is a single number
         int portNum = Integer.parseInt(port);
         if (portNum < 1 || portNum > 65535) {
-          throw new NumberFormatException("Invalid port number " + port);
+          logger.warn("Port number '" + port + "' is out of range (1-65535)");
         }
-        result.put(port + "/tcp", EmptyStruct.get());
+        result.add(port);
 
       } else if (port.matches("\\d+-\\d+")) {
         // Port is a range
         List<String> range = Splitter.on('-').splitToList(port);
         int min = Integer.parseInt(range.get(0));
         int max = Integer.parseInt(range.get(1));
 
+        // Reverse range if configured as 'max-min' instead of 'min-max'
+        if (min > max) {
+          int swapTemp = max;
+          max = min;
+          min = swapTemp;
+        }
+
         // Make sure range is valid (min-max, within port range)
-        if (min > max || min < 1 || max > 65535) {
-          throw new NumberFormatException("Invalid port range " + port);
+        if (min < 1 || max > 65535) {
+          logger.warn("Port range '" + port + "' exceeds normal port range (1-65535)");
         }
 
         for (int portNum = min; portNum <= max; portNum++) {
-          result.put(portNum + "/tcp", EmptyStruct.get());
+          result.add("" + portNum);
         }
 
       } else {
         // Port is neither a single number nor a range
-        throw new NumberFormatException("Invalid port number " + port);
+        logger.warn("Port '" + port + "' is not a port number");
```

Comment:
This should probably throw an error instead since the user wrote an invalid configuration.